include/network.h

```diff
@@ -33,6 +33,18 @@
     #include "socketserver.h"
 #endif
 
+    // For now, just a centralized location for the port numbers for our
+    // various services. Someday these might be configurable.
+    // This could be an enum class, but the static_cast<int> at the
+    // callers is ugly.
+    enum NetworkPort

```

Comment:
Why aren't we making this an enum class to keep our global namespace clean? It seems you use the enum name prefix already, anyway.

Reply:
> Besides, Look in globals.h and tell me really care about clean global namespaces. :-)

We do. That's why we've been grunting at globals.h for ages already, and I've spent quite a lot of time cleaning and bundling global variables in the past - I have my eye on a few more that may undergo the same treatment.

> We could use std::to_underlying, but that's C++23, which we're not exactly prepared to embrace yet. (I'm still working on 20...)

There actually is a "constexpr to_value()" in jsonserializer.h that does the same, which we could move to types.h - which is already included in network.h anyway.

But I agree, it's suboptimal A vs suboptimal B. I'm fine leaving it like this to reduce the hassle. 

> Now that I found the source for RemoteDebug (and files with leading dot are, by definition, hidden)

:) From a file system perspective on UN*X, yes. That's not how I interpreted your statement.

> If you nominate adding the "23" in there with an entry in here, we'll make RemoteDebug depend on our network.h (that may or may not play out well...) 

No, I don't like doing that directly. What we could do is turn the current telnet port define in RemoteDebug into an #ifndef/#define, and define it just above the RemoteDebug include.
To be totally honest on the other hand, 23 has been the telnet port for a very long time, so I'm actually also fine with just leaving it there.

Reply:
My preference for inclusion of the telnet port was based on our mutual earlier belief that doing so would be pretty straightforward. As I now understand it that conclusion no longer stands, which changes things of course. Reading your explanation we could dodge this by sticking the enum in a different header file that does not introduce circular dependencies, but we'd always still have a situation where a dependency end up depending on a project internal header - which is something I failed to realize the last time we discussed this. That's just doing things the wrong way around.

So yes, I'll review one more time with this knowledge in mind.

Thanks for digging into this and providing this level of clarity on the matter!